### PR TITLE
feat/dump-execution

### LINKED ISSUE #1596: Add an "execution dump" button on the execution overview page.
### Feature description

Add an "execution dump" button on the execution overview page.

This will download the execution object so it can be analyzed for debugging purposes.

### PR DESCRIPTION
**Pull request by brian-mulier-p**
closes #1596

### CODE REVIEW

File: ui/src/components/executions/ExecutionRoot.vue [hunk 1]
```diff
@@ -79,6 +86,11 @@
             },
         },
         methods: {
+            downloadExecution() {
+                const blob = new Blob([JSON.stringify(this.execution)], {type: "application/octet-stream"});
+                const url = window.URL.createObjectURL(blob)
+                Utils.downloadUrl(url, `execution_${this.$route.params.id}.txt`);
```
Comment (Skraye) [inline ui/src/components/executions/ExecutionRoot.vue:92]: Put it in a json file as it's json content and format it if possible for more readability

**Review by Skraye — CHANGES_REQUESTED** [review]

**Comment by loicmathieu** [comment]
I think it would be great to not only download an execution JSON but to also add more debugging information, that's why I ask for dump and not just download.

On the top of my mind: logs and some instance information can be interesting

**Comment by brian-mulier-p** [comment]
> I think it would be great to not only download an execution JSON but to also add more debugging information, that's why I ask for dump and not just download.
> 
> On the top of my mind: logs and some instance information can be interesting

I'm wondering if it is the right place to put it then. Also, I've been struggling with translations for this one so I went for "download" translation in french, any idea what it could be translatted to then ?
Also, logs are already downloadable from the logs page, I'm not sure what to do with this button because it feels like a trash dump ?

**Comment by tchiotludo** [comment]
- first will be I think a simple link to execution from the api
- second a dump generated from the server with a zip containing the execution.json, the log as text, the metrics also, and the flow as yaml
- last: a button on the ui to be able to upload the zip and see the execution you are the user :rocket:

**Pull request merged**